### PR TITLE
refactor(core): refactor resume code in sender

### PR DESCRIPTION
**Pull request by kptkin**
Description
-----------

What does the PR do? Include a concise description of the PR contents.

Refactor resume code so it would be easier to add support for fork/rewind in a follow up PR (#7998)

- [x] I updated CHANGELOG.md, or it's not applicable

Testing
-------
How was this PR tested?

### CODE REVIEW

**Review by dmitryduev — APPROVED** [review]
LGTM!

File: core/internal/runbranch/state_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	assert.Equal(t, r.RunID, r2.RunID)
+	assert.Equal(t, r.Project, r2.Project)
+	assert.Equal(t, r.Entity, r2.Entity)
+	assert.Equal(t, r.DisplayName, r2.DisplayName)
+	assert.Equal(t, r.StartTime, r2.StartTime)
+	assert.Equal(t, r.StorageID, r2.StorageID)
+	assert.Equal(t, r.SweepID, r2.SweepID)
+	assert.Equal(t, r.StartingStep, r2.StartingStep)
+	assert.Equal(t, r.Runtime, r2.Runtime)
+	assert.Equal(t, r.Config, map[string]interface{}{
```
Comment (kptkin) [inline core/internal/runbranch/state_test.go:92]:
```suggestion
	assert.Equal(t, r.Config, map[string]any{
```

File: core/internal/runbranch/state_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+		"test":  "test",
+		"test2": "test2",
+	})
+	assert.Equal(t, r.Summary, map[string]interface{}{
```
Comment (kptkin) [inline core/internal/runbranch/state_test.go:96]:
```suggestion
	assert.Equal(t, r.Summary, map[string]any{
```

File: core/internal/runbranch/state_test.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+		"test":  "test",
+		"test2": "test2",
+	})
+	assert.Equal(t, r.Tags, r2.Tags)
```
Comment (kptkin) [inline core/internal/runbranch/state_test.go:100]: what? you need to check the merged value

Comment (kptkin) [reply core/internal/runbranch/state_test.go:100]: sorry it is same

Comment (dmitryduev) [reply core/internal/runbranch/state_test.go:100]: We do:
```go
// update the tags
if len(other.Tags) > 0 {
    r.Tags = other.Tags
}
```

Comment (kptkin) [reply core/internal/runbranch/state_test.go:100]: yeah sorry

**Pull request merged**